xapian-bindings/csharp/csharp.i

```diff
@@ -236,6 +236,8 @@ namespace Xapian {
 
 }
 
+%warnfilter(842);
```

Comment:
(a) Use `WARN_CSHARP_COVARIANT_RET` not `842` - the named constant is at least some-what self-documenting; the number code means nothing unless the reader digs out the corresponding SWIG documentation.
(b) This will ignore any case where there's a covariant return types - we want to only ignore those we've checked to be OK.